docs: add migration guide from 1.1.x to 1.2.x (#1522)
lightbend/kalix-jvm-sdk: docs/migration-guides.md

### Java SDK

> **From 1.2.x onwards, the Java SDK is now called Java Protobuf SDK.**

Accordingly, the package dependencies were updated accordingly:
- from `kalix-java-sdk`to `kalix-java-sdk-protobuf`
- from `kalix-java-sdk-testkit` to `kalix-java-sdk-protobuf-testkit`.

Thus, in your `pom.xml`, instead of:

```xml
    <dependency>
      <groupId>io.kalix</groupId>
      <artifactId>kalix-java-sdk</artifactId>
      <version>${kalix-sdk.version}</version>
    </dependency>
    <dependency>
      <groupId>io.kalix</groupId>
      <artifactId>kalix-java-sdk-testkit</artifactId>
      <version>${kalix-sdk.version}</version>
      <scope>test</scope>
    </dependency>
```

Use:
```xml
    <dependency>
      <groupId>io.kalix</groupId>
      <artifactId>kalix-java-sdk-protobuf</artifactId>
      <version>${kalix-sdk.version}</version>
    </dependency>
    <dependency>
      <groupId>io.kalix</groupId>
      <artifactId>kalix-java-sdk-protobuf-testkit</artifactId>
      <version>${kalix-sdk.version}</version>
      <scope>test</scope>
    </dependency>
```

That's it!

### Scala SDK

>**From 1.2.x onwards, the Scala SDK is now called Scala Protobuf SDK.**

> No changes required. The upgrade is a simple version bump here.

Reviewer: Correct, we also change the name of the scala artifacts, but...

This the dependency is included by the sbt plugin. So it's transparent for the user.

    </dependency>
    <dependency>
      <groupId>io.kalix</groupId>
      <artifactId>kalix-spring-sdk-testkit</artifactId>
      <version>${kalix-sdk.version}</version>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-test</artifactId>
      <version>3.0.2</version>
      <scope>test</scope>
    </dependency>
```

There should now be:
```xml
    <dependency>
      <groupId>io.kalix</groupId>
      <artifactId>kalix-spring-boot-starter</artifactId>
      <version>${kalix-sdk.version}</version>
    </dependency>
    <dependency>
      <groupId>io.kalix</groupId>
      <artifactId>kalix-spring-boot-starter-test</artifactId>
      <version>${kalix-sdk.version}</version>
      <scope>test</scope>
    </dependency>
```

#### Spring Boot Configurations

The following configurations are no longer needed and should be removed: 
- `spring.main.web-application-type=none` in `application.properties` (feel free to remove the file if that's the only property there)
- in your `Main.java`, you can now remove `@Import(KalixConfiguration.class)`

#### EventSourcedEntity with typed events

> A `EventSourcedEntity<S>` is now `EventSourcedEntity<S, E>` where `E` is the top type for all the different event types such event-sourced entity will be allowed to emit. 

Author: Yeah, all these are under section `### Spring SDK`.. 

Maybe something like would help?
```suggestion
An `EventSourcedEntity<S>` is now `EventSourcedEntity<S, E>` where `E` is the top type for all the different event types such event-sourced entity will be allowed to emit. Although on the protocol-first SDKs mentioned in the previous sections this is taken care of automatically, for the code-first approach you need to provide that type. 
```

Thus, instead of :
```java
public class ShoppingCartEntity extends EventSourcedEntity<ShoppingCart> {
  // command and event handlers ...
}
```
There should be now:
```java
public class ShoppingCartEntity extends EventSourcedEntity<ShoppingCart, ShoppingCartEvent> {
  // command and event handlers ...
}
```

Where an example of this `ShoppingCartEvent`, could be:
```java

> public sealed interface Events {

Reviewer: ```suggestion
public sealed interface ShoppingCartEvent {
```

  @TypeName("item-added")

>   record ItemAdded(ShoppingCart.LineItem item) implements Events { }
> 
>   @TypeName("item-removed")
>   record ItemRemoved(String productId) implements Events { }

Reviewer: ```suggestion
  record ItemAdded(ShoppingCart.LineItem item) implements ShoppingCartEvent { }

  @TypeName("item-removed")
  record ItemRemoved(String productId) implements ShoppingCartEvent { }
```

import kalix.springsdk.annotations.EntityType;
import kalix.springsdk.annotations.EventHandler;
import kalix.springsdk.annotations.GenerateEntityKey;
```

It should now be:
```java
import kalix.javasdk.annotations.EntityKey;
import kalix.javasdk.annotations.EntityType;
import kalix.javasdk.annotations.EventHandler;
import kalix.javasdk.annotations.GenerateEntityKey;
```

Note these are only a few example of the existing annotations but all of them were moved.

#### TestKits

The Entity-specific test kits moved from `kalix.springsdk.testkit.*` to `kalix.javasdk.testkit.*`.

Thus, you will need to replace these:
```java
import kalix.springsdk.testkit.ActionTestkit;
import kalix.springsdk.testkit.EventSourcedTestKit;
import kalix.springsdk.testkit.ValueEntityTestKit;
```

With:

```java
import kalix.javasdk.testkit.ActionTestkit;
import kalix.javasdk.testkit.EventSourcedTestKit;
import kalix.javasdk.testkit.ValueEntityTestKit;
```

Lastly, `KalixIntegrationTestKitSupport` was also moved, so you will need to replace:
```java
import kalix.springsdk.testkit.KalixIntegrationTestKitSupport;
```
With:
```java
import kalix.spring.testkit.KalixIntegrationTestKitSupport;

> ```

Reviewer: There is one more thing to change. 

Users should delete `src/main/resources/application.properties`.

This file contains a setting to tell Spring to not start the webserver. That was needed in order to not start the spring netty.  

This now handled by the `starter` and if this old hack in place, our own server (based on akka-http). Our server is now configured as the default Spring reactive server, but the settings tell Spring to not start one if it finds one.

Author: That is mentioned in line #98 under "Spring Boot Configurations".. let me know what you think

Reviewer: Oh, I missed. Thanks!
